Makefile

```
@@ -45,3 +44,26 @@ tidy::
 	cd pkg/tests && go mod tidy
 	cd testing && go mod tidy
 	go mod tidy
+
+# Update the subtree in pkg/tf2pulumi/internal/tf with the latest code from terraform master
+TFPKGS=backend\|command\|provider-terraform\|provider-simple\|provider-simple-v6\|provisioner-local-exec\|cloud\|builtin\|grpcwrap\|initwd\|earlyconfig\|moduledeps\|plugin\|tfplugin5\|plugin6\|tfplugin6\|communicator\|providercache\|e2e\|legacy\|moduletest\|helper\|repl\|terminal
+pull_terraform::
+	git remote remove terraform || true
+	git remote add -f -t main --no-tags terraform https://github.com/hashicorp/terraform.git
+	git rm -rf ./pkg/tf2pulumi/internal/tf
+	git read-tree --prefix=pkg/tf2pulumi/internal/tf/ -u terraform/main:internal
+	# Remove the parts of Terraform we don't need
+	find ./pkg/tf2pulumi/internal/tf -mindepth 1 -maxdepth 1 -type d -regex ".*/tf/\($(TFPKGS)\)" -exec rm -rf {} \;
+	# No need for us to keep any of Terraforms tests
+	find ./pkg/tf2pulumi/internal/tf -type f -regex ".*_test.go" -delete
+	# Update the .go file import paths
+	find ./pkg/tf2pulumi/internal/tf -type f -name "*.go" -exec sed -i -e 's|github.com/hashicorp/terraform/internal|github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tf2pulumi/internal/tf|g' {} \;
```

Note that sed behaves differently on macOS vs Linux (GNU sed vs BSD sed).
Specifically, on macOS, `-i` _requires_ an argument and if one isn't specified, it'll treat `-e` as  the backup file suffix, which will generate "foo-e" files for every "foo" file, which will get checked in and committed below.

We could try to make the sed cross-platform, or we could use perl which behaves the same everywhere.
(Also, this can also use the `+` instead of `;` regardless of sed or perl.)

```suggestion
	find ./pkg/tf2pulumi/internal/tf -type f -name "*.go" -exec perl -p -i -e 's|github.com/hashicorp/terraform/internal|github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tf2pulumi/internal/tf|g' {} '+'
```